### PR TITLE
Revert "[CGData][MachineOutliner] Global Outlining (#90074)"

### PR DESCRIPTION
**Pull request by kyulee-com**
This reverts commit 0f525452896771cc8c579eb362dc7645e38fd0b9.

### CODE REVIEW

File: llvm/include/llvm/ADT/StableHashing.h [hunk 1]
```diff
@@ -53,12 +53,6 @@ inline stable_hash stable_hash_combine(stable_hash A, stable_hash B,
 // Removes suffixes introduced by LLVM from the name to enhance stability and
 // maintain closeness to the original name across different builds.
 inline StringRef get_stable_name(StringRef Name) {
-  // Return the part after ".content." that represents contents.
-  auto [P0, S0] = Name.rsplit(".content.");
-  if (!S0.empty())
-    return S0;
-
```
Comment (kuhar) [inline llvm/include/llvm/ADT/StableHashing.h:60]: Just a drive-by comment: It would be nice to have changes to core llvm headers like this in a separate PR, so that reverting doesn't cause everyone to rebuild a large portion of the build tree.

Comment (kyulee-com) [reply llvm/include/llvm/ADT/StableHashing.h:60]: Sure. I will follow this up!

**Comment by kyulee-com** [comment]
Looks like I can make a forward fix with https://github.com/llvm/llvm-project/pull/108037 instead of this revert.
